TestDirectResponse unit test failure

https://prow.istio.io/view/gcs/istio-prow/logs/istio-unit-tests-master/4494

The objects are semantically the same but the order of header operations is different on some runs:
```go
--- FAIL: TestDirectResponse (0.00s) session_test.go:227: route directive mismatch in multiline set-cookie header 
'&RouteDirective{
	RequestHeaderOperations:[]HeaderOperation{},
	ResponseHeaderOperations:[]HeaderOperation{
		HeaderOperation{Name:Set-Cookie,Value:c1=1; Secure; HttpOnly;,Operation:APPEND,},
		HeaderOperation{Name:Set-Cookie,Value:c2=2,Operation:APPEND,},
		HeaderOperation{Name:Location,Value:URL,Operation:REPLACE,},
        },
	DirectResponseCode:301,DirectResponseBody:,}' != 
'&RouteDirective{
	RequestHeaderOperations:[]HeaderOperation{},
	ResponseHeaderOperations:[]HeaderOperation{
		HeaderOperation{Name:Location,Value:URL,Operation:REPLACE,},
		HeaderOperation{Name:Set-Cookie,Value:c1=1; Secure; HttpOnly;,Operation:APPEND,},
		HeaderOperation{Name:Set-Cookie,Value:c2=2,Operation:APPEND,},
	},
	DirectResponseCode:301,DirectResponseBody:,}'
```
The equality will be changed to check the content, not order, of header operations

--- a/mixer/pkg/runtime/dispatcher/session_test.go
+++ b/mixer/pkg/runtime/dispatcher/session_test.go
@@ -25,6 +25,7 @@ import (
 
 	tpb "istio.io/api/mixer/adapter/model/v1beta1"
 	mixerpb "istio.io/api/mixer/v1"
+	v1 "istio.io/api/mixer/v1"
 	descriptor "istio.io/api/policy/v1beta1"
 	"istio.io/istio/mixer/pkg/adapter"
 	"istio.io/pkg/attribute"
@@ -224,9 +225,32 @@ func TestDirectResponse(t *testing.T) {
 	for _, tc := range testcases {
 		s := &session{}
 		s.handleDirectResponse(tc.status, tc.response)
-		if !reflect.DeepEqual(s.checkResult.RouteDirective, tc.directive) {
+		if s.checkResult.RouteDirective.DirectResponseCode != tc.directive.DirectResponseCode ||
+			s.checkResult.RouteDirective.DirectResponseBody != tc.directive.DirectResponseBody ||
+			!equalHeaderOperations(s.checkResult.RouteDirective.RequestHeaderOperations, tc.directive.RequestHeaderOperations) ||
+			!equalHeaderOperations(s.checkResult.RouteDirective.ResponseHeaderOperations, tc.directive.ResponseHeaderOperations) {
 			t.Fatalf("route directive mismatch in %s '%+v' != '%+v'", tc.desc, s.checkResult.RouteDirective, tc.directive)
 		}
-
 	}
+}
+
+func equalHeaderOperations(actual, expected []v1.HeaderOperation) bool {
+	if len(actual) != len(expected) {
+		return false
+	}
+	delta := make(map[v1.HeaderOperation]int)
+
+	for _, ex := range expected {
+		delta[ex]++
+	}
+	for _, h := range actual {
+		if _, ok := delta[h]; !ok {
+			return false
+		}
+		delta[h]--
+		if delta[h] == 0 {
+			delete(delta, h)
+		}
+	}
+	return len(delta) == 0
 }

--- a/mixer/pkg/runtime/dispatcher/session_test.go
+++ b/mixer/pkg/runtime/dispatcher/session_test.go
@@ -25,7 +25,6 @@ import (
 
 	tpb "istio.io/api/mixer/adapter/model/v1beta1"
 	mixerpb "istio.io/api/mixer/v1"
-	v1 "istio.io/api/mixer/v1"
 	descriptor "istio.io/api/policy/v1beta1"
 	"istio.io/istio/mixer/pkg/adapter"
 	"istio.io/pkg/attribute"
@@ -233,11 +232,11 @@ func TestDirectResponse(t *testing.T) {
 	}
 }
 
-func equalHeaderOperations(actual, expected []v1.HeaderOperation) bool {
+func equalHeaderOperations(actual, expected []mixerpb.HeaderOperation) bool {
 	if len(actual) != len(expected) {
 		return false
 	}
-	delta := make(map[v1.HeaderOperation]int)
+	delta := make(map[mixerpb.HeaderOperation]int)
 
 	for _, ex := range expected {
 		delta[ex]++

--- a/mixer/pkg/runtime/dispatcher/session_test.go
+++ b/mixer/pkg/runtime/dispatcher/session_test.go
@@ -132,7 +132,6 @@ func TestSession_EnsureParallelism(t *testing.T) {
 }
 
 func TestDirectResponse(t *testing.T) {
-	t.Skip("https://github.com/istio/istio/issues/15932")
 	testcases := []struct {
 		desc      string
 		status    rpc.Status

